src/not-so-smart/dune

```diff
@@ -8,8 +8,7 @@
  (name smart)
  (public_name git.nss.smart)
  (modules smart filter capability state protocol)
- (libraries git.nss.pkt-line stdlib-shims result rresult ipaddr domain-name
-   astring fmt))
+ (libraries git.nss.pkt-line result rresult ipaddr domain-name astring fmt))
```

Comment:
it's likely that result can as well be dropped here (result is compat with < 4.03)

Reply:
Unfortunately, it seems that `rresult` depends to `result`. We chosen to use `(implicit_transitive_deps false)` in this project to be sure to be explicit about all dependencies. So if I remove `result`, `dune` complains that `Rresult.t` is not compatible with `Stdlib.result`.

For more information, you can see this issue: https://github.com/mirage/irmin/pull/1024